nearcore/src/runtime/mod.rs

```diff
@@ -770,11 +767,27 @@ impl RuntimeAdapter for NightshadeRuntime {
             / (runtime_config.wasm_config.ext_costs.gas_cost(ExtCosts::storage_write_value_byte)
                 + runtime_config.wasm_config.ext_costs.gas_cost(ExtCosts::storage_read_value_byte));
 
-        while total_gas_burnt < transactions_gas_limit
-            && total_size < size_limit
-            && transactions.len() < new_receipt_count_limit
-            && !time_limit_reached()
-        {
+        // Add new transactions to the result until some limit is hit or the transactions run out.
+        loop {
+            if total_gas_burnt >= transactions_gas_limit {
+                result.limited_by = Some(PrepareTransactionsLimit::Gas);
+                break;
+            }
+            if total_size >= size_limit {
+                result.limited_by = Some(PrepareTransactionsLimit::Size);
+                break;
+            }
+            if result.transactions.len() >= new_receipt_count_limit {
+                result.limited_by = Some(PrepareTransactionsLimit::ReceiptCount);
+                break;
+            }
+            if let Some(tlimit) = &time_limit {
```

Comment:
nit: `if let Some(time_limit) = &time_limit` would be more rusty way to do it

Reply:
My instinct is to avoid shadowing variables at all cost, it's easier on the brain when one name corresponds to one object. And I guess I had many bad experiences with bugs caused by shadowing things.
But okay, I'll rename `tlimit` to`time_limit` if that's more rusty.

Reply:
Generally speaking I agree with you, in this particular case, one could argue it's still the same thing, the only difference is the type and that is checked by the compiler.